solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -1202,26 +1203,26 @@ public void initializeMetrics(SolrMetricsContext parentContext, String scope) {
     parentContext.gauge(() -> isClosed() ? parentContext.nullString() : getIndexDir(), true, "indexDir", Category.CORE.toString());
     parentContext.gauge(() -> isClosed() ? parentContext.nullNumber() : getIndexSize(), true, "sizeInBytes", Category.INDEX.toString());
     parentContext.gauge(() -> isClosed() ? parentContext.nullString() : NumberUtils.readableSize(getIndexSize()), true, "size", Category.INDEX.toString());
-    if (coreContainer != null) {
```

Comment:
Since we do reference coreContainer on like 1215 should we keep this check?

Reply:
I removed the check for CoreContainer nullness as it seemed impossible for it to be null.  SolrCores hang off of a CoreContainer.